pages/_app.js

```diff
@@ -0,0 +1,35 @@
+import {
+  createInstance,
+  enums,
+  OptimizelyProvider,
+  setLogLevel,
+} from "@optimizely/react-sdk";
+
+let optimizely = null;
+function MyApp({ Component, pageProps }) {
+  if (!optimizely) {
+    optimizely = createInstance({
+      sdkKey: process.env.NEXT_PUBLIC_SDK_KEY,
+    });
+  }
+
+  setLogLevel(enums.LOG_LEVEL.ERROR);
+
+  const modifiedProps = { ...pageProps, optimizely };
+
+  return (
+    <>
+      <OptimizelyProvider
```

Comment:
Can you generalize this and the same thing you are doing with SSR provider into one HOC component